### PR TITLE
Count students registered in game

### PR DESCRIPTION
**Pull request by aroquev00**
This is going to need some refactoring later as we discussed on Pod Sync regarding the three sequential reads I do when the user registers in a game. For now, when a user wants to enter a game, the user inserts the gameInstanceId and the User's Id is added to the GameInstance's record on Firebase. The servlets verifies whether or not it is already registered in the GameInstance and if not, it registers the User and increases the member counter which the teacher can see.

### CODE REVIEW

File: src/main/webapp/student/scripts/enter-game.js [hunk 1]
```diff
@@ -1,20 +1,80 @@
...
+      return false;
+    } else {
+      return true;
+    }
+  });
 }
+
+// Send the request to join the Game Instance
+function joinGameInstance(gameInstanceId) {
+  firebase.auth().currentUser.getIdToken(/* forceRefresh */ true).then(function(idToken) {
```
[the remark this answers is not in the document]
Comment (aroquev00) [reply src/main/webapp/student/scripts/enter-game.js:53]: I think it just makes sure that the Token is "fresh" whenever it's clicked, to assure its validity.

File: src/main/java/com/google/sps/daos/DatabaseUserDao.java [hunk 2]
```diff
@@ -55,12 +55,65 @@ public void verifyUserInFirestore(String idToken) {
...
+    } catch (InterruptedException e) {
+      // TODO Auto-generated catch block
+      e.printStackTrace();
+    } catch (ExecutionException e) {
+      // TODO Auto-generated catch block
+      e.printStackTrace();
+    }
+
+    // Add the active GameInstance to the User's entry in "Users" collection
+    ApiFuture<WriteResult> userActiveGameInstanceFuture = firestoreDb.collection("users").document(userId).update("activeGameInstanceId", gameInstanceId);
```
[the remark this answers is not in the document]
Comment (aroquev00) [reply src/main/java/com/google/sps/daos/DatabaseUserDao.java:93]: You're right, it's not, removed!

File: src/main/java/com/google/sps/daos/DatabaseUserDao.java [hunk 3]
```diff
@@ -115,5 +118,15 @@ private void registerUserInGameInstance(DocumentReference userInGameInstanceDocR
 
     userInGameInstanceDocRef.set(docData);
   }
+
+  private void addOneToMembersCounter(DocumentReference gameInstanceDocRef) {
```
[the remark this answers is not in the document]
Comment (aroquev00) [reply src/main/java/com/google/sps/daos/DatabaseUserDao.java:122]: Exactly.

File: src/main/webapp/student/scripts/enter-game.js [hunk 4]
```diff
@@ -1,20 +1,80 @@
...
+    fetch('/joinGameInstance', {
+      method: 'POST',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({idToken: idToken, gameInstanceId: gameInstanceId})
+    }).then(() => {
+      window.location.replace(window.location.origin + "/student/play-game.html");
```
[the remark this answers is not in the document]
Comment (aroquev00) [reply src/main/webapp/student/scripts/enter-game.js]: Right, I already pass the gameInstanceId parameter, and will receive it and work with it in play-game.js in the next PR (#117)

**Comment by aroquev00** [comment]
I added the ability to not make the first query to firestore by checking if the query parameter includes the information about the Game Instance the user wants to enter, should save some times in most cases!

File: src/main/java/com/google/sps/daos/GameDao.java [hunk 5]
```diff
@@ -4,4 +4,5 @@
 public interface GameDao {
   public abstract boolean createNewGame(Game newGame);
   public abstract Game getGame(String id);
+  public abstract String getNextQuestionId(String gameId, String currentQuestionId);
```
Comment (thefredo1000) [inline src/main/java/com/google/sps/daos/GameDao.java:7]: We may have some merge conflicts because everyone is working on this file

Comment (thefredo1000) [reply src/main/java/com/google/sps/daos/GameDao.java:7]: And on `DatabaseGameDao.java` but you get it

**Pull request merged**